dashboard/src/actions/catalog.test.tsx

```diff
@@ -165,6 +166,47 @@ describe("provision", () => {
     await store.dispatch(provisionCMD);
     expect(store.getActions()).toEqual(expectedActions);
   });
+
+  it("filters the submitted parameters if they are not required", async () => {
+    const schema: JSONSchema6 = {
+      required: ["a", "b"],
+    };
+    const params = {
+      a: 1,
+      b: {
+        c: [],
+      },
+      f: {
+        g: [],
+        h: {
+          i: "",
+        },
+      },
+      j: {
+        k: {
+          l: "m",
+        },
+      },
+    };
+    // It omits "f" because it's empty but not "b" because it's required
```

Comment:
Thanks for explanation.

What's the difference between sending empty required parameters and not sending them at all? Wouldn't not sending the empty parameters regardless of being required or not have the same effect?

Reply:
right, my comment was related to not to send the parameters if they are empty in any case. The difference is that you still send them if they are required params but I am suggesting not to send them even in that case. I might be missing something though.

Reply:
okay, yes, it's true that it should be safe to not send anything that is empty.